shell/platform/linux/BUILD.gn

```diff
@@ -114,21 +114,45 @@ test_fixtures("flutter_linux_fixtures") {
 }
 
 executable("flutter_linux_unittests") {
+  public = _public_headers
```

Comment:
Why does an executable need to re-vend headers? Nothing should be depending on headers via this target.

Reply:
Without this I get errors like:

```
FAILED: build.ninja 
../../flutter/third_party/gn/gn --root=../.. -q --check gen .
ERROR at //flutter/shell/platform/linux/fl_basic_message_channel.cc:5:11: Include not allowed.
#include "flutter/shell/platform/linux/public/flutter_linux/fl_basic_message_channel.h"
          ^---------------------------------------------------------------------------
It is not in any dependency of
  //flutter/shell/platform/linux:flutter_linux_unittests
The include file is in the target(s):
  //flutter/shell/platform/linux:flutter_linux
  //flutter/shell/platform/linux:publish_headers_linux
at least one of which should somehow be reachable.
```

Is there another method to allow this?